### PR TITLE
Drop DDT from API content view filter tests

### PR DESCRIPTION
**Pull request by Ichimonji10**
Test results:

    $ nosetests tests/foreman/api/test_contentviewfilter.py
    ...........................S..
    ----------------------------------------------------------------------
    Ran 30 tests in 201.065s

    OK (SKIP=1)

### CODE REVIEW

**Comment by omaciel** [comment]
ack

File: tests/foreman/api/test_contentviewfilter.py [hunk 1]
```diff
@@ -346,11 +353,12 @@ def test_update_cvf_with_new_name(self, new_name):
         cvf = entities.RPMContentViewFilter(
             content_view=self.content_view,
         ).create()
-        cvf.name = new_name
-        self.assertEqual(cvf.update(['name']).name, new_name)
+        for name in _positive_data():
+            with self.subTest(name):
+                cvf.name = name
+                self.assertEqual(cvf.update(['name']).name, name)
```
Comment (Ichimonji10) [inline tests/foreman/api/test_contentviewfilter.py]: I've changed the format of all of the `test_update_*` tests. Formerly, the tests had this structure:

    for name in names:
        create a content view filter
        update the content view filter
        make assertions about the updated content view filter

Now, the tests have this structure:

    create a content view filter
    for name in names:
        update content view filter
        make assertions about updated content view filter

As a result, the update_* tests will now run much more quickly. More gains could be made by defining a class specifically for the update tests:

```python
class ContentViewFilterUpdateTestCase(APITestCase):
    # set some class attribute to disable multi-threading

    @classmethod
    def setUpClass(cls):
        cls.content_view = ...
```

So only one content view filter would need to be made for the entire collection of update tests. But I haven't made that change here.

Comment (omaciel) [inline tests/foreman/api/test_contentviewfilter.py]: approved

**Comment by elyezer** [comment]
ACK

**Comment by Ichimonji10** [comment]
Related to #2769.

**Pull request merged**